tst/Pester.RSpec.Configuration.ts.ps1

```diff
@@ -623,11 +627,247 @@ i -PassThru:$PassThru {
                 }
                 Output = @{
                     StackTraceVerbosity = "Something"
+                    CIFormat            = 'None'
                 }
             }
 
             $r = Invoke-Pester -Configuration $c
             $r.Containers[0].Blocks[0].ErrorRecord[0] | Verify-Equal "Unsupported level of stacktrace output 'Something'"
         }
     }
+
+    b "Output.CIFormat" {
+        t "Output.CIFormat is AzureDevops when Auto(default) and TF_BUILD are set" {
+            $c = [PesterConfiguration] @{
+                Run    = @{
+                    ScriptBlock = { }
+                    PassThru    = $true
+                }
+                Output = @{
+                    Verbosity = "None"
+                }
+            }
+
+            $previousTfBuildVariable = $env:TF_BUILD
+            $previousGithubActionsVariable = $env:GITHUB_ACTIONS
+
+            $env:TF_BUILD = $true
+            $env:GITHUB_ACTIONS = $false
+
+            $r = Invoke-Pester -Configuration $c
+            $r.Configuration.Output.CIFormat.Value | Verify-Equal "AzureDevops"
+
+            $env:TF_BUILD = $previousTfBuildVariable
+            $env:GITHUB_ACTIONS = $previousGithubActionsVariable
+        }
+
+        t "Output.CIFormat is AzureDevops when Auto(manually set) and TF_BUILD are set" {
+            $c = [PesterConfiguration] @{
+                Run    = @{
+                    ScriptBlock = { }
+                    PassThru    = $true
+                }
+                Output = @{
+                    Verbosity = "None"
+                    CIFormat  = "Auto"
+                }
+            }
+
+            $previousTfBuildVariable = $env:TF_BUILD
+            $previousGithubActionsVariable = $env:GITHUB_ACTIONS
+
+            $env:TF_BUILD = $true
+            $env:GITHUB_ACTIONS = $false
+
+            $r = Invoke-Pester -Configuration $c
+            $r.Configuration.Output.CIFormat.Value | Verify-Equal "AzureDevops"
+
+            $env:TF_BUILD = $previousTfBuildVariable
```

Comment:
Those would not be set back if the test fails. You should use try finally to wrap the call to Invoke-Pester.

Reply:
Ah good catch, thanks for spotting that 👍 . I've wrapped those calls in `try/finally`.